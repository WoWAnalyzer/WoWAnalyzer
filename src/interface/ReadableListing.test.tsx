import renderer from 'react-test-renderer';

import ReadableListing from './ReadableListing';

const Dummy = () => <div />;

describe('ReadableList', () => {
  it("doesn't mutate a single item", () => {
    expect(
      renderer
        .create(
          <ReadableListing>
            <Dummy />
          </ReadableListing>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
  it('merges two items with an "and"', () => {
    expect(
      renderer
        .create(
          <ReadableListing>
            <Dummy />
            <Dummy />
          </ReadableListing>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
  it('merges three items with a "," and an "and"', () => {
    expect(
      renderer
        .create(
          <ReadableListing>
            <Dummy />
            <Dummy />
            <Dummy />
          </ReadableListing>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
  it('merges four items with two "," and an "and"', () => {
    expect(
      renderer
        .create(
          <ReadableListing>
            <Dummy />
            <Dummy />
            <Dummy />
            <Dummy />
          </ReadableListing>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
