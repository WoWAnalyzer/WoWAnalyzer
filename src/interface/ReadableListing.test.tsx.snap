// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReadableList > doesn't mutate a single item 1`] = `<div />`;

exports[`ReadableList > merges four items with two "," and an "and" 1`] = `
[
  <div />,
  ", ",
  <div />,
  ", ",
  <div />,
  " and ",
  <div />,
]
`;

exports[`ReadableList > merges three items with a "," and an "and" 1`] = `
[
  <div />,
  ", ",
  <div />,
  " and ",
  <div />,
]
`;

exports[`ReadableList > merges two items with an "and" 1`] = `
[
  <div />,
  " and ",
  <div />,
]
`;
